plugins/scanners/dos/src/main/kotlin/DOSResultParser.kt

```diff
@@ -32,91 +40,51 @@ internal fun generateSummary(startTime: Instant, endTime: Instant, jsonString: S
         issues
     )
 }
-private fun getLicenseFindings(result: JsonNode): Set<LicenseFinding> {
-    val licenseFindings = mutableListOf<LicenseFinding>()
-    val licenses = result["licenses"] ?: return emptySet()
-
-    licenses.forEach { licenseNode ->
-        val license = licenseNode["license"].asText()
-        val location = licenseNode["location"]
-        val path = location["path"].asText()
-        val startLine = location["start_line"]?.asInt()
-        val endLine = location["end_line"]?.asInt()
-        val score = licenseNode["score"]?.asDouble()
-        if (startLine != null && endLine != null && score != null) {
-            licenseFindings.add(
-                LicenseFinding(
-                    license,
-                    TextLocation(
-                        path,
-                        startLine,
-                        endLine
-                    ),
-                    score.toFloat()
-                )
-            )
-
-        } else {
-            licenseFindings.add(
-                LicenseFinding(
-                    license,
-                    TextLocation(
-                        path,
-                        1,
-                        1
-                    ),
-                    100.0f
-                )
-            )
-        }
+private fun getLicenseFindings(result: JsonObject): Set<LicenseFinding> {
+    val licenses = result["licenses"]?.jsonArray ?: return emptySet()
+
+    return licenses.mapTo(mutableSetOf()) {
+        val licenseNode = it.jsonObject
+
+        val license = licenseNode.getValue("license").jsonPrimitive.content
+        val location = licenseNode.getValue("location").jsonObject
+
+        val path = location.getValue("path").jsonPrimitive.content
+        val startLine = location.getValue("start_line").jsonPrimitive.int
+        val endLine = location.getValue("end_line").jsonPrimitive.int
+        val score = licenseNode.getValue("score").jsonPrimitive.float
+
+        LicenseFinding(license, TextLocation(path, startLine, endLine), score)

```

Comment:
@Etsija, @lamppu, my assumption here is that the original code that returns a text location with line 1 and a score of 100 is not really needed, but was a fallback to return "something" if any of `startLine`, `endLine`, or `score` is `null` (which should never happen, IMO). Please correct me if I'm wrong here.